### PR TITLE
Added NavLink routing to layout.

### LINKED ISSUE #3: 3. As a user, I want to be able to navigate to all the pages in the application by clicking the links in the nav bar
## Summary

Users need to be able to navigate between the pages of our application in order to access all of its features. There is a navigation bar in the `Layout.jsx` component, but it's not quite feature-complete: the links in the nav element don't actually allow the user to navigate from page to page.

## Acceptance criteria
- [x] The `nav` element is added to the `Layout` component
- [x] Links to the "Home", "List", and "Add item" pages of the app are In the `nav` element
- [x] The links all function as expected using  the `NavLink` component from `react-router-dom`

## Notes
- The [official docs for the `NavLink` component](https://reactrouter.com/en/6.14.2/components/nav-link) will come in very handy!
- While a standard anchor element (`<a>`) _could_ allow a user to navigate from page to page, the `NavLink` component adds additional behavior that is essential to our app functioning correctly!

### PR DESCRIPTION
**Pull request by DevinaG007**
## Description

Updated the navigation links to use the `<NavLink>` element from React Router to fix link functionality.

## Related Issue

closes #3

## Acceptance Criteria

- [x] The `nav` element is added to the `Layout` component
- [x] Links to the "Home", "List", and "Add item" pages of the app are In the `nav` element
- [x] The links all function as expected using  the `NavLink` component from `react-router-dom`

## Type of Changes
`Code Refactoring`

### Before

Not a UI feature, before and after screenshots are not included.

### After

Not a UI feature, before and after screenshots are not included.

## Testing Steps / QA Criteria
Clicking **Home** leads to homepage
Clicking **List** leads to list page
Clicking **Manage List** leads to manage-list page

### CODE REVIEW

**Review by sdmisra — APPROVED** [review]
This looks good to me!

**Review by etiry — APPROVED** [review]
Agreed, looks good!

**Comment by DevinaG007** [comment]
Great, I'll pass it on to our mentors to review before we merge the PR.

File: src/views/Layout.jsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Outlet } from 'react-router-dom';
+import { Outlet, NavLink } from 'react-router-dom';
 
 import './Layout.css';
```
Comment (cafloyd) [inline src/views/Layout.jsx:3]: Looks beautiful!!! 👏 👏 👏 

@DevinaG007 @andiedoescode Now that you've implemented these changes, you can also go ahead and delete the TODO comment here in Layout.jsx from lines 7-13 to keep things nice and tidy! ✅

**Review by cafloyd — APPROVED** [review]
Great work @andiedoescode and @DevinaG007 !! Well done with your first issue and PR!!

**Pull request merged**